### ISSUE TITLE
[Bug] The transcripts info contained in the spatialData points doesn't store the information of the matching new cell_ID with the new segmentations when running CELLPOSE

### ISSUE THREAD

**Issue by munizmom** [gustaveroussy/sopa#293]
## Description
The transcripts info stored in the spatial data--points--transcripts, after performing new segmentations with cellpose in a. Xenium dateset and even running assign.transcripts_to_cell doesn't store the segmentation polygons cell_IDs and instead or store the old segmentation cellID or a novel numeric one created after  running assign.transcripts_to_cell
## Reproducing the issue
Providing the script
## Expected behavior
the cell_IDs or the new IDs created after running assign.transcripts_to_cell should match the cellID stored in the cell boundaries table file.
## System
 - OS: [e.g. Linux] Macos
 - Python version [e.g. 3.9.7] 3.10.18

[issue1_sopa_transcripts_MM.txt](https://github.com/user-attachments/files/21781146/issue1_sopa_transcripts_MM.txt)

<img width="1118" height="611" alt="Image" src="https://github.com/user-attachments/assets/2e633cac-821a-46f1-8da4-a9dae72ca004" />

<details>
<summary>Dependencies versions</summary>
<br>
Paste here what 'pip list' gives you.
</details>Package                       Version
----------------------------- -----------
absl-py                       2.3.1
affine                        2.4.0
aiobotocore                   2.23.0
aiohappyeyeballs              2.6.1
aiohttp                       3.12.14
aioitertools                  0.12.0
aiosignal                     1.4.0
alabaster                     1.0.0
alphashape                    1.3.1
anndata                       0.11.4
annotated-types               0.7.0
app-model                     0.3.2
appdirs                       1.4.4
appnope                       0.1.4
array-api-compat              1.12.0
asciitree                     0.3.3
asttokens                     3.0.0
astunparse                    1.6.3
async-timeout                 5.0.1
attrs                         25.3.0
babel                         2.17.0
bioio                         3.0.0
bioio-base                    3.0.0
botocore                      1.38.27
build                         1.2.2.post1
cachey                        0.2.1
cellpose                      4.0.6
certifi                       2025.7.9
charset-normalizer            3.4.2
click                         8.2.1
click-log                     0.4.0
click-plugins                 1.1.1.2
cligj                         0.7.2
cloudpickle                   3.1.1
colorcet                      3.1.0
comm                          0.2.2
ComSeg                        1.8.5
contourpy                     1.3.2
csbdeep                       0.8.1
cycler                        0.12.1
dask                          2024.11.2
dask-expr                     1.1.19
dask-image                    2024.5.3
datashader                    0.18.1
debugpy                       1.8.14
decorator                     5.2.1
distributed                   2024.11.2
docrep                        0.3.2
docstring_parser              0.16
docutils                      0.21.2
exceptiongroup                1.3.0
executing                     2.2.0
fasteners                     0.19
fastremap                     1.17.1
filelock                      3.18.0
fill_voids                    2.1.0
flatbuffers                   25.2.10
flexcache                     0.3
flexparser                    0.4
FlowIO                        1.4.0
fonttools                     4.58.5
freetype-py                   2.5.1
frozenlist                    1.7.0
fsspec                        2025.5.1
gast                          0.6.0
geopandas                     1.1.1
google-pasta                  0.2.0
grandalf                      0.8
grpcio                        1.73.1
h5py                          3.14.0
HeapDict                      1.0.1
hf-xet                        1.1.5
hsluv                         5.0.4
huggingface-hub               0.33.3
idna                          3.10
igraph                        0.11.9
imagecodecs                   2025.3.30
imageio                       2.37.0
imagesize                     1.4.1
importlib_metadata            8.7.0
in-n-out                      0.2.1
inflect                       7.5.0
ipykernel                     6.29.5
ipython                       8.37.0
jedi                          0.19.2
Jinja2                        3.1.6
jmespath                      1.0.1
joblib                        1.5.1
jsonschema                    4.24.0
jsonschema-specifications     2025.4.1
jupyter_client                8.6.3
jupyter_core                  5.8.1
keras                         3.10.0
kiwisolver                    1.4.8
lazy_loader                   0.4
legacy-api-wrap               1.4.1
leidenalg                     0.10.2
libclang                      18.1.1
lightning                     2.5.2
lightning-utilities           0.14.3
llvmlite                      0.44.0
locket                        1.0.0
loguru                        0.7.3
loompy                        3.0.8
lxml                          6.0.0
magicgui                      0.10.1
Markdown                      3.8.2
markdown-it-py                3.0.0
MarkupSafe                    3.0.2
matplotlib                    3.10.3
matplotlib-inline             0.1.7
matplotlib-scalebar           0.9.0
mdurl                         0.1.2
ml_dtypes                     0.5.1
more-itertools                10.7.0
mpmath                        1.3.0
msgpack                       1.1.1
multidict                     6.6.3
multipledispatch              1.0.0
multiscale_spatial_image      2.0.2
namex                         0.1.0
napari                        0.5.5
napari-brightness-contrast    0.1.8
napari-console                0.1.3
napari-matplotlib             3.0.0
napari-plugin-engine          0.2.0
napari-spatialdata            0.5.6
napari-svg                    0.2.1
napari-tools-menu             0.1.20
natsort                       8.4.0
nest-asyncio                  1.6.0
netgraph                      4.13.2
networkx                      3.4.2
novae                         0.2.4
npe2                          0.7.9
numba                         0.61.2
numcodecs                     0.13.1
numpy                         1.26.4
numpy-groupies                0.11.3
numpydoc                      1.9.0
ome-types                     0.6.1
ome-zarr                      0.11.1
omnipath                      1.0.12
opencv-python                 4.11.0.86
opencv-python-headless        4.11.0.86
opt_einsum                    3.4.0
optree                        0.16.0
packaging                     25.0
pandas                        2.3.1
param                         2.2.1
parso                         0.8.4
partd                         1.4.2
patsy                         1.0.1
pexpect                       4.9.0
pillow                        11.3.0
PIMS                          0.7
Pint                          0.24.4
pip                           25.1.1
platformdirs                  4.3.8
pooch                         1.8.2
prompt_toolkit                3.0.51
propcache                     0.3.2
protobuf                      5.29.5
psutil                        7.0.0
psygnal                       0.14.0
ptyprocess                    0.7.0
pure_eval                     0.2.3
pyarrow                       20.0.0
pyconify                      0.2.1
pyct                          0.5.0
pydantic                      2.11.7
pydantic-compat               0.1.2
pydantic_core                 2.33.2
pydantic-extra-types          2.10.5
Pygments                      2.19.2
pynndescent                   0.5.13
pyogrio                       0.11.0
PyOpenGL                      3.1.9
pyparsing                     3.2.3
pyproj                        3.7.1
pyproject_hooks               1.2.0
PyQt5                         5.15.11
PyQt5-Qt5                     5.15.17
PyQt5_sip                     12.17.0
pyqtgraph                     0.13.7
python-dateutil               2.9.0.post0
python-louvain                0.16
pytorch-lightning             2.5.2
pytz                          2025.2
PyYAML                        6.0.2
pyzmq                         27.0.0
qtconsole                     5.6.1
QtPy                          2.4.3
rasterio                      1.4.3
readfcs                       2.0.1
rectangle-packer              2.0.4
referencing                   0.36.2
requests                      2.32.4
rich                          14.0.0
rioxarray                     0.19.0
roifile                       2025.5.10
rpds-py                       0.26.0
rtree                         1.4.0
s3fs                          2025.5.1
safetensors                   0.5.3
scanpy                        1.11.3
scikit-image                  0.25.2
scikit-learn                  1.7.0
scikit-misc                   0.5.1
scipy                         1.15.3
seaborn                       0.13.2
segment-anything              1.0
semver                        3.0.4
session-info2                 0.1.2
setuptools                    80.9.0
shapely                       2.1.1
shellingham                   1.5.4
six                           1.17.0
slicerator                    1.1.0
snowballstemmer               3.0.1
sopa                          2.1.0
sortedcontainers              2.4.0
spatial_image                 1.2.1
spatialdata                   0.4.0
spatialdata-io                0.2.0
spatialdata-plot              0.2.10
Sphinx                        8.1.3
sphinxcontrib-applehelp       2.0.0
sphinxcontrib-devhelp         2.0.0
sphinxcontrib-htmlhelp        2.1.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          2.0.0
sphinxcontrib-serializinghtml 2.0.0
squidpy                       1.6.5
stack-data                    0.6.3
stardist                      0.9.1
statsmodels                   0.14.5
superqt                       0.7.5
sympy                         1.14.0
tblib                         3.1.0
tensorboard                   2.19.0
tensorboard-data-server       0.7.2
tensorflow                    2.19.0
tensorflow-io-gcs-filesystem  0.37.1
termcolor                     3.1.0
texttable                     1.7.0
threadpoolctl                 3.6.0
tifffile                      2025.5.10
tiffslide                     2.5.1
tinycss2                      1.4.0
toml                          0.10.2
tomli                         2.2.1
tomli_w                       1.2.0
toolz                         1.0.0
torch                         2.7.1
torch-geometric               2.6.1
torchmetrics                  1.7.4
torchvision                   0.22.1
tornado                       6.5.1
tqdm                          4.67.1
traitlets                     5.14.3
trimesh                       4.7.1
typeguard                     4.4.4
typer                         0.16.0
typing_extensions             4.14.1
typing-inspection             0.4.1
tzdata                        2025.2
umap-learn                    0.5.9.post2
urllib3                       2.5.0
validators                    0.35.0
vispy                         0.14.3
wcwidth                       0.2.13
webencodings                  0.5.1
Werkzeug                      3.1.3
wheel                         0.45.1
wrapt                         1.17.2
xarray                        2024.11.0
xarray-dataclasses            1.9.1
xarray-datatree               0.0.14
xarray-schema                 0.0.3
xarray-spatial                0.4.0
xsdata                        24.3.1
yarl                          1.20.1
zarr                          2.18.3
zict                          3.0.0
zipp                          3.23.0
